app.py

```
@@ -1,4 +1,75 @@
+    
+    # Fetch all courses
+    cursor.execute("SELECT id, name FROM courses")
+    courses = cursor.fetchall()
+    
+    conn.close()
+    
+    return render_template('index.html', selected_date='', no_data=False, courses=courses)
+
+
+@app.route('/attendance', methods=['GET', 'POST'])
+def attendance():
```

Ruff (0.8.2)

Missing return type annotation for public function `attendance`

(ANN201)